### PR TITLE
Fixed issues in C code examples and added modern C standard idioms (C99 & C11).

### PR DESCRIPTION
**Pull request by schlupa**
Small changes in the "Programming in D for C Programmers" page.

### CODE REVIEW

File: ctod.dd [hunk 1]
```diff
@@ -208,14 +208,14 @@ $(H3 <a name="nans">Dealing with NANs in floating point compares</a>)
...
-       C compiler is an exception, DM's compilers do check for NAN operands).
+       is `NAN`, and few C compilers check for it (the Digital Mars
+       C compiler is an exception, DM's compilers do check for `NAN` operands).
 
 $(CCODE
 #include &lt;math.h&gt;
 
 if (isnan(x) || isnan(y))
-    result = FALSE;
+    result = false;
```
Comment (schlupa) [inline ctod.dd]: Since C99 bool type is defined in the language and `true` and `false` are keyword.

File: ctod.dd [hunk 2]
```diff
@@ -275,8 +276,8 @@ $(H3 <a name="arrayloop">Looping through an array</a>)
 
 $(H4 The C Way)
 $(P
-       The array length is defined separately, or a clumsy
-       sizeof() expression is used to get the length.)
+       The array length is defined separately, or a clumsy and error prone
```
Comment (schlupa) [inline ctod.dd]: the very common idiom `sizeof array/sizeof array[0]` breaks when the `array` expression decays to a pointer as arrays often do in C.

File: ctod.dd [hunk 3]
```diff
@@ -718,23 +719,23 @@ Sometimes, it's nice to control the layout of a struct with nested structs and u
...
 
 $(CCODE
 struct Foo
 {
     int i;
-    union Bar
+    union
     {
-        struct Abc { int x; long y; } _abc;
+        struct { int x; long y; } abc;
```
Comment (schlupa) [inline ctod.dd]: The struct and union tags were never required in C.

File: ctod.dd [hunk 4]
```diff
@@ -233,9 +233,10 @@ result = (x < y);        // false if x or y is nan
 $(H3 <a name="assert">Asserts are a necessary part of any good defensive coding strategy</a>)
 
 $(H4 The C Way)
-$(P C doesn't directly support assert, but does support __FILE__
-and __LINE__ from which an assert macro can be built. In fact,
-there appears to be practically no other use for __FILE__ and __LINE__.)
+$(P C doesn't directly support assert in the language, but does define a
```
Comment (BBasile) [inline ctod.dd]: this paragraph is not closed properly

**Pull request merged**